packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1439,3 +1479,239 @@ export async function saveFile(doc: vscode.TextDocument, datasetProvider: api.IZ
+    if (!Array.isArray(clipboardContent) && clipboardContent.memberName) {
+        return pasteMember(node, datasetProvider);
+    }
+
+    await api.Gui.withProgress(
+        {
+            location: vscode.ProgressLocation.Window,
+            title: localize("paste.dataSet.InPrg", "Copying File(s)"),
+        },
+        async function copyDsMember() {
+            for (const content of clipboardContent) {
+                if (content.memberName) {
+                    try {
+                        await ZoweExplorerApiRegister.getMvsApi(node.getProfile()).copyDataSetMember(
+                            { dsn: content.dataSetName, member: content.memberName },
+                            { dsn: node.getLabel().toString(), member: content.memberName }
+                        );
+                    } catch (err) {
+                        api.Gui.errorMessage(err.message);
+                        return;
+                    }
+                }
+            }
+            datasetProvider.refreshElement(node);
+            vscode.env.clipboard.writeText("");
+        }
+    );
+}
+
+/**
+ * download given dataset node
+ *
+ * @export
+ * @param {ZoweDatasetNode} node - node to be downloaded
+ */
+export async function downloadDs(node: ZoweDatasetNode) {
+    const profile = node.getProfile();
+    let lbl: string;
+    switch (true) {
+        case contextually.isFavorite(node):
+        case contextually.isSessionNotFav(node.getParent()):
+            lbl = node.label as string;
+            break;
+        case contextually.isFavoritePds(node.getParent()):
+        case contextually.isPdsNotFav(node.getParent()):
+            lbl = node.getParent().getLabel().toString() + "(" + node.getLabel().toString() + ")";
+            break;
+        default:
+            vscode.window.showErrorMessage(localize("download.invalidNode", "downloadDataset() called with invalid node."));
+            throw Error(localize("download.invalidNode", "downloadDataset() called with invalid node. "));
+    }
+    const filePath = getDocumentFilePath(lbl, node);
+    return ZoweExplorerApiRegister.getMvsApi(profile).getContents(lbl, {
+        file: filePath,
+        returnEtag: true,
+        encoding: profile.profile.encoding,
+    });
+}
+
+/**
+ * copies given sequential dataset nodes
+ *
+ * @export
+ * @param {ZoweDatasetNode[]} nodes - nodes to be copied
+ */
+export async function copySequentialDatasets(nodes: ZoweDatasetNode[]) {
+    await _copyProcessor(nodes, "ps", async (node: ZoweDatasetNode, dsname: string, replace: shouldReplace) => {
+        const lbl = node.getLabel().toString();
+
+        await api.Gui.withProgress(
+            {
+                location: vscode.ProgressLocation.Window,
+                title: localize("paste.dataSet.InPrg", "Copying File(s)"),
+            },
+            () => {
+                return ZoweExplorerApiRegister.getMvsApi(nodes[0].getProfile()).copyDataSet(lbl, dsname, null, replace === "replace");
```

Comment:
If copyDataSet hasn't been picked up by an extender yet we should have a check for this otherwise user will see messages like below. 
We shouldn't expect extenders to add 'Not Supported' like I see in the FTP package here, it should be handled by the ZE extension like we have in the past where if the particular API didn't exist we either 
1. handle it the old way if this is a new way
2. return the error of this action not being supported if the new api doesn't exist

![Screen Shot 2023-02-27 at 11 40 19 AM](https://user-images.githubusercontent.com/49491949/221627294-883254ff-2d62-4df9-876a-b1c3554893f3.png)

Reply:
And error message will appear indicating that `Copying datasets is not supported`
However, that leaves the question of whether we want to prevent the `allocateLike` from happening or not.

As it stands, I believe the new Copy/Paste experience is a little better than before.

Do we want to address the AllocateLike (and possibly other misbehaviors that we haven't found yet) in this PR?

(Personal opinion) 
I have considered going with a similar approach as in #2082 where we rely on the clipboard instead of performing a complex, multi-step, error-prone, user interaction for such functionality. 
However, some design decisions need to be discussed. For example, the USS file system may not require allocation of resources.

Reply:
I think Allocate like can be a future PR